### PR TITLE
Add Reset() to cache

### PR DESCRIPTION
**Pull request by wendigo**
* Add ```Reset()``` method to ```BigCache``` (reset all shards)
* Add ```Reset()``` method to BytesQueue (reset queue to initial state)
* Add ```Len()``` method to ```BigCache``` (number of entries)
* Move cacheShard to it's own file
* Move config logic

Addresses https://github.com/allegro/bigcache/issues/23

### CODE REVIEW

File: iterator.go [hunk 1]
```diff
@@ -47,7 +44,7 @@ func (e EntryInfo) Value() []byte {
 
 // EntryInfoIterator allows to iterate over entries in the cache
 type EntryInfoIterator struct {
-	sync.Mutex
+	mutex         sync.Mutex
```
Comment (janisz) [inline iterator.go]: What's the reason to change it?

File: queue/bytes_queue.go [hunk 2]
```diff
@@ -39,17 +40,28 @@ type queueError struct {
...
 	}
 }
 
+// Reset removes all entries from queue
+func (q *BytesQueue) Reset() {
+	// Just reset indexes
+	q.tail = leftMarginIndex
+	q.head = leftMarginIndex
+	q.rightMargin = leftMarginIndex
+	q.capacity = q.initialCapacity
```
Comment (janisz) [inline queue/bytes_queue.go]: Do we need to set it with reset? Capacity shouldn't be changed because we didn't shrink `q.array`

File: shard.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package bigcache
```
Comment (janisz) [inline shard.go]: Nice move :)

File: iterator.go (hunk 1 above)
Comment (wendigo) [inline iterator.go]: It was exposed as public Mutex field

**Pull request merged**